src/components/SearchableMultiSelect/SearchableMultiSelect.jsx

```diff
@@ -243,6 +252,32 @@ const SearchableMultiSelect = createClass({
 		});
 	},
 
+	handleSelectAll(event) {
+		// This is needed otherwise clicking the checkbox will double fire this
+		// event _and_ the `handleDropMenuSelect` handler
+		const {
+			props: { selectedIndices, onSelect },
+			state: { flattenedOptionsData },
+		} = this;
+
+		event.stopPropagation();
+
+		const visibleOptions = _.reject(
```

Comment:
unfortunately this is duplicated from `renderOptions`, I couldn't figure out a clean way to dedupe. I think at some point we'll want to do a deeper refactor and move a lot of this logic out to selectors.